app/controllers/equipment_models_controller.rb

```diff
@@ -202,4 +203,47 @@ def generate_calendar_resource
   def calendar_name_method
     :reserver
   end
+
+  def calculate_availability # rubocop:disable all
+    # get start and end dates
+    @start_date = Time.zone.today.beginning_of_week(:sunday)
+    @end_date = (Time.zone.today + 1.month).end_of_week(:sunday)
+
+    # hack-y, we have a proper scope in master for this
+    reservations =
+      (Reservation.for_eq_model(@equipment_model).active
+      .where('start_date <= ? and due_date >= ?', @end_date, @start_date) + \
+      Reservation.for_eq_model(@equipment_model).overdue).uniq
+    max_avail = @equipment_model.equipment_items.active.count
+
+    @avail_data = []
+    (@start_date..@end_date).map do |date|
+      count = 0
+      # also hack-y, we need to fix these methods
+      reservations.each do |res|
+        if res.overdue && res.checked_out?
+          count += 1
+        elsif res.start_date <= date && res.due_date >= date
+          count += 1
+        end
+      end
+
+      availability = max_avail - count
```

Comment:
My feeling is that it belongs in the controller since we're using queries on multiple tables (i.e. `reservations` and `equipment_items`) and putting everything together. Moreover, we're performing an action that's specific to a user request (e.g. displaying the calendar on the `show` page), so that further justifies keeping it in the controller (in my mind). What do you think?